From 111090805062caf53da1bdabd15ee6fe62b8a66a Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Thu, 2 Apr 2015 16:41:35 +0800
Subject: [PATCH 10/10] ov2722: lower mipi frequency for 1080P resolution

The mipi frequency for 1080P is over 360MHz per lane which
is out of the range recommended in sensor spec.

Change-Id: Ibf2b891b5801a2394d40a1f3e2e3b2bea3253de2
Tracked-On: #JIMINAN-26480
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/49235
Approver: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-by: Shuguang Gong <shuguang.gong@intel.com>
Build: IMIN (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/ov2722.h              | 53 ++++++++++++----------
 1 file changed, 28 insertions(+), 25 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
index 283a037..41e0064 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.h
@@ -916,21 +916,21 @@ static struct ov2722_reg const ov2722_1M3_30fps[] = {
 };
 
 static struct ov2722_reg const ov2722_1080p_30fps[] = {
-	{OV2722_8BIT, 0x3021, 0x03}, /* For stand wait for
-				a whole frame complete.(vblank) */
+	{OV2722_8BIT, 0x3021, 0x03}, /* For stand wait for a whole
+					frame complete.(vblank) */
 	{OV2722_8BIT, 0x3718, 0x10},
 	{OV2722_8BIT, 0x3702, 0x24},
 	{OV2722_8BIT, 0x373a, 0x60},
 	{OV2722_8BIT, 0x3715, 0x01},
 	{OV2722_8BIT, 0x3703, 0x2e},
-	{OV2722_8BIT, 0x3705, 0x10},
+	{OV2722_8BIT, 0x3705, 0x2b},
 	{OV2722_8BIT, 0x3730, 0x30},
 	{OV2722_8BIT, 0x3704, 0x62},
 	{OV2722_8BIT, 0x3f06, 0x3a},
 	{OV2722_8BIT, 0x371c, 0x00},
 	{OV2722_8BIT, 0x371d, 0xc4},
 	{OV2722_8BIT, 0x371e, 0x01},
-	{OV2722_8BIT, 0x371f, 0x0d},
+	{OV2722_8BIT, 0x371f, 0x28},
 	{OV2722_8BIT, 0x3708, 0x61},
 	{OV2722_8BIT, 0x3709, 0x12},
 	{OV2722_8BIT, 0x3800, 0x00},
@@ -946,9 +946,9 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x380a, 0x04},
 	{OV2722_8BIT, 0x380b, 0x44}, /* V output size: 1092 */
 	{OV2722_8BIT, 0x380c, 0x08},
-	{OV2722_8BIT, 0x380d, 0xd4}, /* H timing: 2260 */
+	{OV2722_8BIT, 0x380d, 0x14}, /* H timing: 2068 */
 	{OV2722_8BIT, 0x380e, 0x04},
-	{OV2722_8BIT, 0x380f, 0xdc}, /* V timing: 1244 */
+	{OV2722_8BIT, 0x380f, 0x5a}, /* V timing: 1114 */
 	{OV2722_8BIT, 0x3810, 0x00},
 	{OV2722_8BIT, 0x3811, 0x03}, /* H window offset: 3 */
 	{OV2722_8BIT, 0x3812, 0x00},
@@ -957,7 +957,7 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x3821, 0x06}, /*  mirror */
 	{OV2722_8BIT, 0x3814, 0x11},
 	{OV2722_8BIT, 0x3815, 0x11},
-	{OV2722_8BIT, 0x3612, 0x0b},
+	{OV2722_8BIT, 0x3612, 0x4b},
 	{OV2722_8BIT, 0x3618, 0x04},
 	{OV2722_8BIT, 0x3a08, 0x01},
 	{OV2722_8BIT, 0x3a09, 0x50},
@@ -967,11 +967,14 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x3a0e, 0x03},
 	{OV2722_8BIT, 0x4520, 0x00},
 	{OV2722_8BIT, 0x4837, 0x1b},
+	{OV2722_8BIT, 0x3000, 0xff},
+	{OV2722_8BIT, 0x3001, 0xff},
+	{OV2722_8BIT, 0x3002, 0xf0},
 	{OV2722_8BIT, 0x3600, 0x08},
 	{OV2722_8BIT, 0x3621, 0xc0},
-	{OV2722_8BIT, 0x3632, 0xd2}, /* added for power opt */
-	{OV2722_8BIT, 0x3633, 0x23},
-	{OV2722_8BIT, 0x3634, 0x54},
+	{OV2722_8BIT, 0x3632, 0x53}, /* added for power opt */
+	{OV2722_8BIT, 0x3633, 0x63},
+	{OV2722_8BIT, 0x3634, 0x24},
 	{OV2722_8BIT, 0x3f01, 0x0c},
 	{OV2722_8BIT, 0x5001, 0xc1},
 	{OV2722_8BIT, 0x3614, 0xf0},
@@ -983,7 +986,7 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x4005, 0x08},
 	{OV2722_8BIT, 0x404f, 0x84},
 	{OV2722_8BIT, 0x4051, 0x00},
-	{OV2722_8BIT, 0x5000, 0xcf}, /* manual 3a */
+	{OV2722_8BIT, 0x5000, 0xcd}, /* manual 3a */
 	{OV2722_8BIT, 0x301d, 0xf0}, /* enable group hold */
 	{OV2722_8BIT, 0x3a18, 0x00},
 	{OV2722_8BIT, 0x3a19, 0x80},
@@ -994,7 +997,7 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x5185, 0xb0},
 	{OV2722_8BIT, 0x370c, 0x0c},
 	{OV2722_8BIT, 0x3035, 0x00},
-	{OV2722_8BIT, 0x3036, 0x2c}, /* 422.4 MHz */
+	{OV2722_8BIT, 0x3036, 0x24}, /* 345.6 MHz */
 	{OV2722_8BIT, 0x3037, 0xa1},
 	{OV2722_8BIT, 0x303e, 0x19},
 	{OV2722_8BIT, 0x3038, 0x06},
@@ -1008,7 +1011,7 @@ static struct ov2722_reg const ov2722_1080p_30fps[] = {
 	{OV2722_8BIT, 0x3a1e, 0x30},
 	{OV2722_8BIT, 0x3a11, 0x90},
 	{OV2722_8BIT, 0x3a1f, 0x10},
-	{OV2722_8BIT, 0x3503, 0x07}, /* manual 3a */
+	{OV2722_8BIT, 0x3011, 0x22},
 	{OV2722_8BIT, 0x3500, 0x00},
 	{OV2722_8BIT, 0x3501, 0x3F},
 	{OV2722_8BIT, 0x3502, 0x00},
@@ -1155,17 +1158,17 @@ struct ov2722_resolution ov2722_res_preview[] = {
 		.desc = "ov2722_1080P_30fps",
 		.width = 1932,
 		.height = 1092,
-		.pix_clk_freq = 85,
+		.pix_clk_freq = 69,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 2260,
-		.lines_per_frame = 1244,
+		.pixels_per_line = 2068,
+		.lines_per_frame = 1114,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
 		.regs = ov2722_1080p_30fps,
-		.mipi_freq = 422400,
+		.mipi_freq = 345600,
 	},
 };
 #define N_RES_PREVIEW (ARRAY_SIZE(ov2722_res_preview))
@@ -1207,17 +1210,17 @@ struct ov2722_resolution ov2722_res_still[] = {
 		.desc = "ov2722_1080P_30fps",
 		.width = 1932,
 		.height = 1092,
+		.pix_clk_freq = 69,
 		.fps = 30,
-		.pix_clk_freq = 85,
 		.used = 0,
-		.pixels_per_line = 2260,
-		.lines_per_frame = 1244,
+		.pixels_per_line = 2068,
+		.lines_per_frame = 1114,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
 		.regs = ov2722_1080p_30fps,
-		.mipi_freq = 422400,
+		.mipi_freq = 345600,
 	},
 };
 #define N_RES_STILL (ARRAY_SIZE(ov2722_res_still))
@@ -1274,17 +1277,17 @@ struct ov2722_resolution ov2722_res_video[] = {
 		.desc = "ov2722_1080P_30fps",
 		.width = 1932,
 		.height = 1092,
+		.pix_clk_freq = 69,
 		.fps = 30,
-		.pix_clk_freq = 85,
 		.used = 0,
-		.pixels_per_line = 2260,
-		.lines_per_frame = 1244,
+		.pixels_per_line = 2068,
+		.lines_per_frame = 1114,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
 		.regs = ov2722_1080p_30fps,
-		.mipi_freq = 422400,
+		.mipi_freq = 345600,
 	},
 };
 #define N_RES_VIDEO (ARRAY_SIZE(ov2722_res_video))
-- 
1.9.1

